lib/bundle-validator.js

```diff
@@ -106,6 +107,119 @@ function validateThemeConfiguration(callback) {
     callback(null, true);
 }
 
+/**
+ * Find translatable strings
+ * @param {object} schema
+ * @param {function} callback
+ */
+function getTranslatableStrings(schema, callback) {
+    var trackedKeys = ['name', 'content', 'label', 'settings', 'options'];
+
+    schema.forEach(element => {
+        Object.keys(element).forEach(key => {
+            const value = element[key];
+
+            if (trackedKeys.indexOf(key) === -1) {
+                return;
+            }
+
+            if (Array.isArray(value)) {
+                return getTranslatableStrings(value, callback);
+            }
+
+            callback(value);
+        });
+    });
+}
+
+/**
+ * Find i18n keys in schema
+ * @param {object} schema
+ * @return {array}
+ */
+function getI18nKeys(schema) {
+    var keys = [];
+
+    getTranslatableStrings(schema, (value) => {
+        if (
+            value
+            && keys.indexOf(value) === -1
+            && /^i18n\./.test(value)
+        ) {
+            keys.push(value);
+        }
+    });
+
+    return keys;
+}
+
+/**
+ * Find missed i18n keys in schemaTranslations
+ * @param {array} keys
+ * @param {object} translations
+ * @return {array}
+ */
+function findMissedKeys(keys, translations) {
+    var translationsKeys = Object.keys(translations);
+    var missingKeys = [];
+
+    keys.forEach(key => {
```

Comment:
I think we may be able to refactor this into just using a `.filter`.

`Object.keys(translations).filter(key => translationsKeys.includes(key));`